db2i_file.cc

```diff
@@ -376,9 +376,9 @@ void db2i_table::filenameToTablename(const char* in, char* out, size_t outlen)
     if (!part4)
       part4 = strend(in);
   }
-  
-  memcpy(temp, part1, min(outlen, part2 - part1));
-  temp[min(outlen-1, part2-part1)] = 0;
+  size_t num_bytes = std::min(outlen, static_cast<size_t>(part2 - part1));
+  memcpy(temp, part1, std::min(outlen, num_bytes));
+  temp[std::min(outlen-1, num_bytes)] = 0;
```

Comment:
```suggestion
  size_t num_bytes = std::min(outlen-1, static_cast<size_t>(part2 - part1));
  memcpy(temp, part1, num_bytes);
  temp[num_bytes] = 0;
```

In the memcpy line, outlen has already been used to initialize num_bytes, no need to min against it again. In fact, once the min has been set there's no need to do any further calculations. The only change needed is to reserve one byte from the buffer length for the null terminator.